controllers/tutorial.js

```diff
@@ -121,11 +113,97 @@ const findTutorial = async (req, res) => {
   }
 };
 
+// Users api's
+const registerUsers = async (req, res) => {
+  try {
+    const resultvalidated = await joiuserSchema.validateAsync(req.body);
+    // check if user is already registered
+    const emailExist = await user.findOne({ email: resultvalidated.email });
+    if (emailExist) {
+      return res.status(400).send('Email already exists');
+    }
+    // hash the password
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(resultvalidated.password, salt);
+    const userdb = await user({
+      firstname: resultvalidated.firstname,
+      lastname: resultvalidated.lastname,
+      email: resultvalidated.email,
+      password: hashedPassword,
+    });
+    userdb.save();
+    if (userdb) {
+      return res.status(200).json({
+        userdb,
+      });
+    }
+    return res.status(500).send('problem occurs');
```

Comment:
please write a suitable error message, I think you can omit this line as you already have a catch block.